daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/Transaction.scala

```diff
@@ -419,6 +420,18 @@ sealed abstract class HasTxNodes[Nid, +Cid, +Val] {
     acc
   }
 
+  final def submitter: Either[String, Party] =
+    roots.map(nodes(_).requiredAuthorizers) match {
+      case ImmArray() =>
+        Left(s"Empty transaction has no authorizers")
```

Comment:
This is different from what we had before, where we used the empty set of submitters for an empty transaction. Do we not care about empty transactions anymore?

Reply:
This function is just an utility to help adapt old cold to the new API.

The API requires now  you provide the submitter of the original submission.  
Since there is no way to find this information in an empty transaction this code does not help you in this case.

There are two possible workarounds: 
- One use a dummy party for empty transaction
- One do not run the validation on Empty transaction (they are trivially valid)

However I do expect the ledger implementation to have recorded the original submitter and to not have to use this utility function. Indeed to run the authorization check (`AuthorizingTransaction`) one needs the original submitter.

Reply:
That makes sense. Thanks for the explanation.